src/main/java/taxiservice/dao/jdbc/CarDaoJdbcImpl.java

```diff
@@ -0,0 +1,238 @@
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import taxiservice.dao.CarDao;
+import taxiservice.exception.DataProcessingException;
+import taxiservice.lib.Dao;
+import taxiservice.model.Car;
+import taxiservice.model.Driver;
+import taxiservice.model.Manufacturer;
+import taxiservice.util.ConnectionUtil;
+
+@Dao
+public class CarDaoJdbcImpl implements CarDao {
+    public static final String CARS_TAB_NAME = "cars";
+    public static final String CARS_DRIVERS_TAB_NAME = "cars_drivers";
+
+    @Override
+    public Car create(Car car) {
+        String query = "INSERT INTO " + CARS_TAB_NAME + " (model, manufacturer_id) VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement statement = connection
+                        .prepareStatement(query, Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(1, car.getModel());
+            statement.setLong(2, car.getManufacturer().getId());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                car.setId(resultSet.getLong(1));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't create " + car, e);
+        }
+
+        query = "INSERT INTO " + CARS_DRIVERS_TAB_NAME + " (car_id, diver_id) VALUES (?, ?)";
+        for (Driver driver : car.getDrivers()) {
+            try (Connection connection = ConnectionUtil.getConnection();
+                     PreparedStatement statement = connection
+                             .prepareStatement(query, Statement.RETURN_GENERATED_KEYS)) {
+                statement.setLong(1, car.getId());
+                statement.setLong(2, driver.getId());
+                statement.executeUpdate();
+            } catch (SQLException e) {
+                throw new DataProcessingException("Can't create " + car, e);
+            }
+        }
+        return car;
+    }
+
+    @Override
+    public Optional<Car> get(Long id) {
+        Car result = null;
+        String query = "SELECT * FROM " + CARS_TAB_NAME + " c"
+                + " INNER JOIN " + MANUFACTURERS_TAB_NAME + " m"
+                + " ON c.manufacturer_id = m.manufacturer_id"
+                + " WHERE car_id = ? AND c.deleted = false";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                result = newCar(resultSet);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get by id " + id, e);
+        }
+
+        if (result != null) {
+            result.setDrivers(getDrivers(result.getId()));
```

Comment:
you can put this if/else block in try/catch and use the same connection in  method getDrivers()
```suggestion
            result.setDrivers(getDrivers(result.getId()));
```

Reply:
also agree)